### ISSUE TITLE
The .is() method returns unexpected results.

### ISSUE THREAD

**Issue by trusktr** [jquery/jquery#3452]
### Description ###
- What do you expect to happen?
  - Passing an array to the `.is()` method should work when the array contains selector strings, jQuery objects, or native DOM elements references.
- What actually happens?
  - Passing an array to the `.is()` method only works if the array contains references to native DOM elements, but not when it contains selector strings or jQuery objects.
- Which browsers are affected?
  - Probably all.

### Link to test case ###
Here's a fiddle: https://jsfiddle.net/dw5w2d31/. The test cases that don't have expected results are 7, 8, 13, 14. However, test cases 10 and 11 seem to have the correct results only due to coincidence relating to the same reason why the other cases have the unexpected results.

**Comment by trusktr** [comment]
TLDR: If you'll support array input to the `.is()` method, then it should with with selectors, DOM element references, and jQuery objects alike. These tests show that an array input to `.is()` only works with DOM element references.

**Comment by trusktr** [comment]
I added test cases 15, 16, and 17: https://jsfiddle.net/trusktr/9cawnxsk/. They show that `toArray()` works.

**Comment by trusktr** [comment]
Maybe, for best portability, it should work with any level of nested arrays, see test case 18. https://jsfiddle.net/trusktr/9cawnxsk/

**Comment by gibson042** [comment]
This behavior is documented at http://api.jquery.com/is/ , and I believe the signature is quite complex enough. We probably shouldn't have even mentioned `$collection.is( arrayOfElements )`, since it's redundant with `$collection.is( $otherCollection )`. If you want to test against an array that might include non-elements, I recommend converting it to a jQuery collection: https://jsfiddle.net/dw5w2d31/1/

**Comment by dmethvin** [comment]
I agree, it just confuses the signature to start accepting more different things. The Array case no doubt works because it has a `.length` property and we wanted to accept jQuery collections.

**Comment by timmywil** [comment]
Agreed, it's cleaner and more explicit for the user to convert the array. We don't want to make the API even more magical by allowing mixed type arrays.
